crates/ethjson/src/state/test.rs

```diff
@@ -86,16 +88,27 @@ pub struct MultiTransaction {
 
 impl MultiTransaction {
     /// Build transaction with given indexes.
-    pub fn select(&self, indexes: &PostStateIndexes) -> Transaction {
-        Transaction {
+    pub fn select(&self, indexes: &PostStateIndexes) -> TypedTransaction {
+        let transaction = Transaction {
             data: self.data[indexes.data as usize].clone(),
             gas_limit: self.gas_limit[indexes.gas as usize].clone(),
             gas_price: self.gas_price.clone(),
             nonce: self.nonce.clone(),
             secret: self.secret.clone(),
             to: self.to.clone(),
             value: self.value[indexes.value as usize].clone(),
+        };
+
+        if let Some(access_lists) = self.access_lists.as_ref() {
+            if let Some(access_list) = access_lists[indexes.data as usize].clone() {
```

Comment:
check the length of `access_lists` , if the index is more than the length create it as legacy tx.

Reply:
Fixed